### ISSUE TITLE
Allow using <Controller> outside of react-hook-form

### ISSUE THREAD

**Issue by Smona** [react-hook-form/react-hook-form#784]
**Is your feature request related to a problem? Please describe.**
We are currently building an inputs library, and one of our goals is to make inputs work inside a react-hook-form, or standalone as one-off inputs. I just migrated our controlled inputs over to use Controller, because 4.x has really improved integration of controlled components (thank you!). However, I had to use some ugly conditional logic to determine whether to wrap the input in a controller, because if the Controller is rendered outside of a form context, i get the following error:

```
Uncaught [TypeError: Cannot destructure property `defaultValues` of 'undefined' or 'null'
```

**Describe the solution you'd like**
It would be really nice if the Controller checked whether it was in a form context, and gracefully rendered its children without modification when it isn't.

**Describe alternatives you've considered**
The alternative is to render the input element and do something along the lines of:

```jsx
const { form } = useFormContext()
const input = <MyInput {...} />
return form ? (
  <Controller>{input}</Controller>
) : input
```

This solution works, but is uglier. In the interest of flexibility and "just working", it would be better if this behavior was encapsulated in the library instead.

**Additional context**
n/a

**Comment by bluebill1049** [comment]
strange i believe we are actually doing the check. can you produce a codesandbox for such issue.

**Comment by bluebill1049** [comment]
are you using controller outside of `useForm`?

**Comment by Smona** [comment]
[Here's an example of the error I'm seeing](https://codesandbox.io/s/little-rgb-jei5k?fontsize=14&hidenavigation=1&theme=dark). Not the exact same error, but the same problem.

The idea here is that one should be able to write custom components using the Controller without having to worry about them erroring out if they're used outside of useForm

**Comment by Smona** [comment]
I think tacking ` || {}` onto: https://github.com/react-hook-form/react-hook-form/blob/8262ca0482294bbf3931ffcc58d46ff6ce6b4ac0/src/controller.tsx#L34 
might do the trick. I can get a PR together tomorrow if you're open to the idea

**Comment by bluebill1049** [comment]
sure send a PR if you have time 👍

**Comment by bluebill1049** [comment]
actually think about it twice, it's probably better to throw an error on this. because controller is not going to work outside the context of useForm.

**Comment by cargallo** [comment]
I think IMHO that the requested feature is outside the scope of Controller component. Controller is and should remain been an adapter to use controlled inputs with useForm and not outside of it context.

**Comment by Smona** [comment]
While I did figure out a way to support this, I realized while writing the PR why it's not a great idea. 

Controller maintains its own state for the value, and assumes that value will be accessed from the `useForm`. Making it equivalent to a standard onChange/value api would require changing its own api to add a way to pass in the current value. At that point, I agree with @cargallo that it would be out of scope.

Thanks for your consideration, and thanks again for the amazing library! [Here](https://github.com/Smona/react-hook-form/blob/support-controller-outside-of-useform/src/controller.tsx) is my solution for future reference.
